### ISSUE TITLE
[FreeBSD 11.4] make error upon building in master: assigning to 'int' from incompatible type 'pthread_t' (aka 'pthread *')

### ISSUE THREAD

**Issue by kadabusha** [SIPp/sipp#517]
make for **master branch fails**, while I was able to build&install **successfully from sipp-3.6.1.tar.gz**
```
[user@freebsd ~/sipp]$ uname -r
11.4-RELEASE-p9
[user@freebsd ~]$ pkg info | grep cmake
cmake-3.19.2                   Cross-platform Makefile generator
[user@freebsd ~]$ git clone https://github.com/SIPp/sipp.git -b master
Cloning into 'sipp'...
remote: Enumerating objects: 58, done.
remote: Counting objects: 100% (58/58), done.
remote: Compressing objects: 100% (56/56), done.
remote: Total 6821 (delta 23), reused 8 (delta 2), pack-reused 6763
Receiving objects: 100% (6821/6821), 5.18 MiB | 12.30 MiB/s, done.
Resolving deltas: 100% (4866/4866), done.
[user@freebsd ~]$ cd sipp
[user@freebsd ~/sipp]$ cmake .
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is Clang 10.0.0
-- The CXX compiler identification is Clang 10.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for endian.h
-- Looking for endian.h - not found
-- Looking for sys/endian.h
-- Looking for sys/endian.h - found
-- Looking for sys/epoll.h
-- Looking for sys/epoll.h - not found
-- Performing Test HAVE_UDP_UH_PREFIX
-- Performing Test HAVE_UDP_UH_PREFIX - Success
-- Looking for le16toh
-- Looking for le16toh - not found
-- Looking for le16toh
-- Looking for le16toh - found
-- Performing Test HAVE_CXX_AUTO_KEYWORD
-- Performing Test HAVE_CXX_AUTO_KEYWORD - Success
-- Performing Test HAVE_CXX_RANGE_LOOP
-- Performing Test HAVE_CXX_RANGE_LOOP - Success
-- Found Git: /usr/local/bin/git (found version "2.30.1")
-- Configuring done
-- Generating done
-- Build files have been written to: /home/user/sipp
[user@freebsd ~/sipp]$ make
Scanning dependencies of target version
[  0%] Built target version
Scanning dependencies of target sipp
[  3%] Building CXX object CMakeFiles/sipp.dir/src/actions.cpp.o
[  6%] Building CXX object CMakeFiles/sipp.dir/src/auth.cpp.o
[  9%] Building CXX object CMakeFiles/sipp.dir/src/call.cpp.o
/home/user/sipp/src/call.cpp:1116:20: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
    } else if (ptr = strchr(msg, ' ')) {
               ~~~~^~~~~~~~~~~~~~~~~~
/home/user/sipp/src/call.cpp:1116:20: note: place parentheses around the assignment to silence this warning
    } else if (ptr = strchr(msg, ' ')) {
                   ^
               (                     )
/home/user/sipp/src/call.cpp:1116:20: note: use '==' to turn this assignment into an equality comparison
    } else if (ptr = strchr(msg, ' ')) {
                   ^
                   ==
1 warning generated.
[ 12%] Building CXX object CMakeFiles/sipp.dir/src/call_generation_task.cpp.o
[ 15%] Building C object CMakeFiles/sipp.dir/src/comp.c.o
[ 18%] Building CXX object CMakeFiles/sipp.dir/src/deadcall.cpp.o
[ 21%] Building CXX object CMakeFiles/sipp.dir/src/infile.cpp.o
[ 25%] Building CXX object CMakeFiles/sipp.dir/src/jlsrtp.cpp.o
[ 28%] Building CXX object CMakeFiles/sipp.dir/src/listener.cpp.o
[ 31%] Building CXX object CMakeFiles/sipp.dir/src/logger.cpp.o
[ 34%] Building C object CMakeFiles/sipp.dir/src/md5.c.o
[ 37%] Building CXX object CMakeFiles/sipp.dir/src/message.cpp.o
[ 40%] Building C object CMakeFiles/sipp.dir/src/milenage.c.o
[ 43%] Building CXX object CMakeFiles/sipp.dir/src/ratetask.cpp.o
[ 46%] Building CXX object CMakeFiles/sipp.dir/src/reporttask.cpp.o
[ 50%] Building C object CMakeFiles/sipp.dir/src/rijndael.c.o
[ 53%] Building CXX object CMakeFiles/sipp.dir/src/rtpstream.cpp.o
/home/user/sipp/src/rtpstream.cpp:177:14: error: assigning to 'int' from incompatible type 'pthread_t' (aka 'pthread *')
    retVal = p;
             ^
/home/user/sipp/src/rtpstream.cpp:191:55: warning: format specifies type 'unsigned long' but the argument has type 'pthread_t' (aka 'pthread *')
      [-Wformat]
        fprintf(debugafile, "TID: %lu %s %d %d %d [", pthread_self(), note, size, extrainfo, moreinfo);
                                  ~~~                 ^~~~~~~~~~~~~~
/home/user/sipp/src/rtpstream.cpp:209:55: warning: format specifies type 'unsigned long' but the argument has type 'pthread_t' (aka 'pthread *')
      [-Wformat]
        fprintf(debugvfile, "TID: %lu %s %d %d %d [", pthread_self(), note, size, extrainfo, moreinfo);
                                  ~~~                 ^~~~~~~~~~~~~~
/home/user/sipp/src/rtpstream.cpp:227:55: warning: format specifies type 'unsigned long' but the argument has type 'pthread_t' (aka 'pthread *')
      [-Wformat]
        fprintf(debugafile, "TID: %lu %s %d %d %d [", pthread_self(), note, size, extrainfo, moreinfo);
                                  ~~~                 ^~~~~~~~~~~~~~
/home/user/sipp/src/rtpstream.cpp:246:46: warning: format specifies type 'unsigned long' but the argument has type 'pthread_t' (aka 'pthread *')
      [-Wformat]
        fprintf(debugafile, "TID: %lu %s\n", pthread_self(), note);
                                  ~~~        ^~~~~~~~~~~~~~
/home/user/sipp/src/rtpstream.cpp:263:55: warning: format specifies type 'unsigned long' but the argument has type 'pthread_t' (aka 'pthread *')
      [-Wformat]
        fprintf(debugvfile, "TID: %lu %s %d %d %d [", pthread_self(), note, size, extrainfo, moreinfo);
                                  ~~~                 ^~~~~~~~~~~~~~
/home/user/sipp/src/rtpstream.cpp:282:46: warning: format specifies type 'unsigned long' but the argument has type 'pthread_t' (aka 'pthread *')
      [-Wformat]
        fprintf(debugvfile, "TID: %lu %s\n", pthread_self(), note);
                                  ~~~        ^~~~~~~~~~~~~~
6 warnings and 1 error generated.
*** Error code 1

Stop.
make[2]: stopped in /usr/home/user/sipp
*** Error code 1

Stop.
make[1]: stopped in /usr/home/user/sipp
*** Error code 1

Stop.
make: stopped in /usr/home/user/sipp
```

**Comment by wdoekes** [comment]
Assuming this is fixed by #518

**Comment by kadabusha** [comment]
Thank you, it worked.
```
[user@freebsd ~/sipp]$ sipp -v

 SIPp v3.7-dev-17-g45e586e-TLS-SCTP-PCAP-RTPSTREAM-RTPCHECK.`
```
